multimedia/x265/Portfile

```diff
@@ -4,7 +4,8 @@ PortSystem          1.0
 
 PortGroup           muniversal 1.0
 PortGroup           cmake 1.1
-PortGroup           compiler_blacklist_versions 1.0
+# https://trac.macports.org/ticket/59246
+PortGroup           stack_check_fix 1.0
```

Comment:
X265 still seg. faults if built with Xcode 11.2, so it should be indicated its not fixed in this version.

Reply:
I am happy to make this change.
Just to verify, however, ` /opt/local/bin/x264` works for me on Xcode 11.2.
Is there some other part that is still broken?

Reply:
Sorry to ask the obvious question, but are you sure you have rebuilt from source without the current blacklist ? If I do this on macOS 10.15 with Xcode 11.2 I get

```
 > x265
zsh: segmentation fault  x265
```

whereas with the blacklist and thus using MP clang 9.0 it is fine.

Reply:
I am reasonable certain I am running on macOS 10.15 and Xcode 11.2.
Do you have the Command Line Tools (CLTs) package installed?
If so, you may not be using Xcode at all.

If the CLTs are installed, x265 [uses them by default](https://github.com/macports/macports-base/blob/aeb6d4386cd92afd0924a69bc8ee6154e651e8b3/src/port1.0/portmain.tcl#L160).
As far as I know, the CLTs are still at a lower version, where the bug still exists.
`/usr/sbin/pkgutil --pkg-info=com.apple.pkg.CLTools_Executables` should verify.

That is why this PR sets `use_xcode yes` if it detects that Xcode has been updated but the CLTs have not been.

As a quick check, you can set `use_xcode yes` in x265 and see if you still get the segmentation fault.

Reply:
```
> /usr/sbin/pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
```

Not sure what that means ?

But, yes, I did previously have the CLT installed so probably I guess still have the older version installed.

Reply:
something seems odd here. Both your two ones of detecting the CLI fail for me, but I believe I do nevertheless have it installed ??

```
Oberon ~/Projects/MacPorts/ports > /usr/sbin/pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
Oberon ~/Projects/MacPorts/ports > /usr/sbin/pkgutil --pkg-info=com.apple.pkg.DeveloperToolsCLI
No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.
Oberon ~/Projects/MacPorts/ports > xcode-select --install
xcode-select: error: command line tools are already installed, use "Software Update" to install updates
```

Reply:
OK, so setting `use_xcode yes` does work for me. 

However, it appears your checks for the CLT are not working in my case. I do have the CLT installed, e.g. using the above changes in the compilation

```
-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk
```

to

```
-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk
```

which is what you would expect. 

So, @MarcusCalhoun-Lopez, any idea why your CLT test is not working in my case ?

Reply:
@MarcusCalhoun-Lopez It looks like during the Xcode 11.2 update the package receipts for the CLT tools package was removed from `/var/db/receipts/` but the CLT SDK itself was not removed, so still available. This scenario I think breaks the logic in your PG here I am afraid ?

Reply:
> @MarcusCalhoun-Lopez It looks like during the Xcode 11.2 update the package receipts for the CLT tools package was removed from `/var/db/receipts/` but the CLT SDK itself was not removed, so still available. This scenario I think breaks the logic in your PG here I am afraid ?

Yes, it does.
Darn.

May I ask a favor?
Could you please run `pkgutil --pkgs | grep CLT` to see if any packages might be more suitable?
I am not sure how else to get the CLTs version information.

Reply:
Sorry, but nothing comes up

```
Oberon ~/Projects/MacPorts/ports >  pkgutil --pkgs | grep CLT
Oberon ~/Projects/MacPorts/ports >
```

Moreover a direct look shows nothing related to the CLT

```
Oberon ~/Projects/MacPorts/ports > ls /var/db/receipts/com.apple.pkg.*
/var/db/receipts/com.apple.pkg.GarageBand_AppStore.bom		/var/db/receipts/com.apple.pkg.Numbers5.bom
/var/db/receipts/com.apple.pkg.GarageBand_AppStore.plist	/var/db/receipts/com.apple.pkg.Numbers5.plist
/var/db/receipts/com.apple.pkg.InstallOS.bom			/var/db/receipts/com.apple.pkg.Numbers6.bom
/var/db/receipts/com.apple.pkg.InstallOS.plist			/var/db/receipts/com.apple.pkg.Numbers6.plist
/var/db/receipts/com.apple.pkg.Keynote6.bom			/var/db/receipts/com.apple.pkg.Pages5.bom
/var/db/receipts/com.apple.pkg.Keynote6.plist			/var/db/receipts/com.apple.pkg.Pages5.plist
/var/db/receipts/com.apple.pkg.Keynote7.bom			/var/db/receipts/com.apple.pkg.Pages6.bom
/var/db/receipts/com.apple.pkg.Keynote7.plist			/var/db/receipts/com.apple.pkg.Pages6.plist
/var/db/receipts/com.apple.pkg.Keynote8.bom			/var/db/receipts/com.apple.pkg.Pages7.bom
/var/db/receipts/com.apple.pkg.Keynote8.plist			/var/db/receipts/com.apple.pkg.Pages7.plist
/var/db/receipts/com.apple.pkg.Keynote9.bom			/var/db/receipts/com.apple.pkg.Pages8.bom
/var/db/receipts/com.apple.pkg.Keynote9.plist			/var/db/receipts/com.apple.pkg.Pages8.plist
/var/db/receipts/com.apple.pkg.Numbers3.bom			/var/db/receipts/com.apple.pkg.Xcode.bom
/var/db/receipts/com.apple.pkg.Numbers3.plist			/var/db/receipts/com.apple.pkg.Xcode.plist
/var/db/receipts/com.apple.pkg.Numbers4.bom			/var/db/receipts/com.apple.pkg.iMovie_AppStore.bom
/var/db/receipts/com.apple.pkg.Numbers4.plist			/var/db/receipts/com.apple.pkg.iMovie_AppStore.plist
Oberon ~/Projects/MacPorts/ports > ls /Library/Receipts/*
/Library/Receipts/InstallHistory.plist						/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsJamPack1.bom
/Library/Receipts/com.apple.pkg.GarageBand_AppStore.bom				/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsJamPack1.plist
/Library/Receipts/com.apple.pkg.GarageBand_AppStore.plist			/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsModernRnB.bom
/Library/Receipts/com.apple.pkg.Keynote6.bom					/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsModernRnB.plist
/Library/Receipts/com.apple.pkg.Keynote6.plist					/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsRemixTools.bom
/Library/Receipts/com.apple.pkg.MAContent10_GarageBandCoreContent.bom		/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsRemixTools.plist
/Library/Receipts/com.apple.pkg.MAContent10_GarageBandCoreContent.plist		/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsRhythmSection.bom
/Library/Receipts/com.apple.pkg.MAContent10_GarageBandCoreContent2.bom		/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsRhythmSection.plist
/Library/Receipts/com.apple.pkg.MAContent10_GarageBandCoreContent2.plist	/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsSymphony.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsChillwave.bom	/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsSymphony.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsChillwave.plist	/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsTechHouse.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsDeepHouse.bom	/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsTechHouse.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsDeepHouse.plist	/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsWorld.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsDubstep.bom		/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsWorld.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsDubstep.plist	/Library/Receipts/com.apple.pkg.Numbers3.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsElectroHouse.bom	/Library/Receipts/com.apple.pkg.Numbers3.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsElectroHouse.plist	/Library/Receipts/com.apple.pkg.Pages5.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsGarageBand.bom	/Library/Receipts/com.apple.pkg.Pages5.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsGarageBand.plist	/Library/Receipts/com.apple.pkg.iMovie_AppStore.bom
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsHipHop.bom		/Library/Receipts/com.apple.pkg.iMovie_AppStore.plist
/Library/Receipts/com.apple.pkg.MAContent10_PremiumPreLoopsHipHop.plist
```

Maybe the safe thing to do is if you do not detect the CLT installed using the commands you use, to force the use of Xcode just in case it is installed but you cannot deduce the version.

Reply:
Yes, I think you are right.
`pkgutil` is unreliable.
Thank you for your help.
I will make changes.

This may help explain a problem I had a while back.
Software Update was not updating the command line tools even though I could see there was a new version available.
If Xcode modifies the receipts, perhaps Software Update did not recognize that the CLTs were installed at all.

Reply:
Just FYI, I just manually downloaded and installed the Xcode 11 CLT from

https://developer.apple.com/download/more/?=command%20line%20tools

and now I get

```
Oberon ~/Projects/MacPorts/ports > pkgutil --pkgs | grep CLT
com.apple.pkg.CLTools_Executables
com.apple.pkg.CLTools_SDK_macOS1015
com.apple.pkg.CLTools_SDK_macOS1014
com.apple.pkg.CLTools_macOS_SDK
```

Maybe you could detect the scenario where pkgutil fails to return anything, but a valid SDK exists in `/Library/Developer/CommandLineTools/SDKs/` and issue some kind of warning, and handle the situation accordingly (i.e. force the use of Xcode SDK).

Reply:
Thank you for helping track this down.
I have modified `cltversion-1.0.tcl` to issue a warning if `pkgutil` fails to find a version.
I have also added a section in the [Wiki](https://trac.macports.org/wiki/ProblemHotlist#reinstall-clt).